### PR TITLE
Address issue #107 and further assorted changes

### PR DESCRIPTION
**Pull request by dvandyk**
- resolves issue #107 
   - [x] implement test cases
 - add class `QuarkFlavorOption`
 - `flavour` -> `flavor`
 - update documentation

### CODE REVIEW

**Comment by dvandyk** [comment]
This PR is ready for review!

File: doc/basics.rst [hunk 1]
```diff
@@ -403,18 +403,18 @@ pairs of strings. Within each pair, we refer to the first element as the
...
+- ``model`` can typically take values such as ``SM`` (for the Standard
+  Model), ``CKM`` (to parametrize each CKM matrix element and fit for
+  absolute value or complex phase), and ``WET`` (to parametrize the
+  Wilson coefficients of the Weak Effective Theory);
+- ``form-factors`` can typically take values that identify a single parametrization
+  (e.g. ``BSZ2015`` for a parametrization used in Bharucha, Straub, Zwicky 2015);
+- ``l`` can typically take values ``e``, ``mu`` and ``tau``;
+- ``q`` can typically take values ``u``, ``d``, ``s``, and ``c``.
 
 Option keys are specific to a process. Presently, there is no way to list
```
Comment (pluegh) [inline doc/basics.rst:419]: That is not true anymore, is it?

File: python/eos/analysis.py [hunk 2]
```diff
@@ -375,21 +375,46 @@ def sample(self, N=1000, stride=5, pre_N=150, preruns=3, cov_scale=0.1, observab
 
     def sample_pmc(self, log_proposal, step_N=1000, steps=10, final_N=5000, rng=np.random.mtrand, return_final_only=True, final_perplexity_threshold=1.0):
         """
-        Return samples of the parameters and log(weights)
+        Return samples of the parameters and log(weights), and a mixture density adapted to the posterior.
```
Comment (pluegh) [inline python/eos/analysis.py:378]: The following lines 380/381 are a better general description. The return value is discussed later. Remove this line?

Comment (dvandyk) [reply python/eos/analysis.py:378]: Agreed!

File: eos/b-decays/b-to-k-nu-nu_TEST.cc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                    { "D",            "s"       },
+                    { "q",            "u"       },
+                    { "I",            "1/2"     }
+                };
+
+                BToPseudoscalarDineutrino d(p, oo);
+
+                const double eps = 1e-8;
+
+                TEST_CHECK_NEARLY_EQUAL(2.05845e-06, d.integrated_branching_ratio( 0.00,  8.00), eps);
```
Comment (mreboud) [inline eos/b-decays/b-to-k-nu-nu_TEST.cc:72]: I find slightly inconsistent to set `eps = 1e-8` and then give number with only 5 decimals, no?

File: eos/b-decays/b-to-kstar-nu-nu_TEST.cc [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    { "D",            "s"       },
+                    { "q",            "d"       },
+                    { "I",            "1/2"     }
+                };
+
+                BToVectorDineutrino d(p, oo);
+
+                const double eps = 1e-10;
+
+                TEST_CHECK_NEARLY_EQUAL(2.84988e-06, d.integrated_branching_ratio( 0.00,  8.00), eps);
```
Comment (mreboud) [inline eos/b-decays/b-to-kstar-nu-nu_TEST.cc:83]: Idem, `1e-10` seems very small. also, I would prefer using a relative error check, it seems more physical to me

File: eos/b-decays/b-to-vec-nu-nu.cc [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+            opt_q(o, options, "q"),
+            opt_I(o, options, "I"),
+            m_B(p["mass::" + _B()], u),
+            tau_B(p["life_time::" + _B()], u),
+            m_V(p["mass::" + _V()], u),
+            g_fermi(p["WET::G_Fermi"], u),
+            alpha_e(p["QED::alpha_e(m_b)"], u),
+            hbar(p["QM::hbar"], u),
+            isospin_factor(_isospin_factor()),
+            mu(p[opt_D.str() + "b" + "nunu::mu"], u),
```
Comment (mreboud) [inline eos/b-decays/b-to-vec-nu-nu.cc:150]: Since the WilsonCoefficient object doesn't depend on `mu`, why do you have this parameter here?

**Review by mreboud — APPROVED** [review]

File: eos/b-decays/b-to-k-nu-nu_TEST.cc (hunk 3 above)
Comment (mreboud) [reply eos/b-decays/b-to-k-nu-nu_TEST.cc:72]: Forget about this, it's not a relative error. I would still prefer to use relative errors

Comment (dvandyk) [reply eos/b-decays/b-to-k-nu-nu_TEST.cc:72]: We can go for a relative error, no problem as long as the values are non-zero.

File: eos/b-decays/b-to-vec-nu-nu.cc (hunk 5 above)
Comment (dvandyk) [reply eos/b-decays/b-to-vec-nu-nu.cc:150]: The matrix elements do depend on it, in general. It's not applicable here, but hadronic matrix element of the scalar and/or tensor operators are explicitly mu dependent.

Comment (mreboud) [reply eos/b-decays/b-to-vec-nu-nu.cc:150]: understood!

File: eos/b-decays/b-to-kstar-nu-nu_TEST.cc (hunk 4 above)
Comment (dvandyk) [reply eos/b-decays/b-to-kstar-nu-nu_TEST.cc:83]: Done.

File: eos/b-decays/b-to-k-nu-nu_TEST.cc (hunk 3 above)
Comment (dvandyk) [reply eos/b-decays/b-to-k-nu-nu_TEST.cc:72]: Done. `1e-4` for all as relative error.

File: doc/basics.rst (hunk 1 above)
Comment (dvandyk) [reply doc/basics.rst]: Done.

**Pull request merged**